clients/drcachesim/tools/opcode_mix.cpp

```diff
@@ -81,30 +81,35 @@ opcode_mix_t::opcode_mix_t(const std::string &module_file_path, unsigned int ver
 {
 }
 
+void
+opcode_mix_t::make_decode_cache(shard_data_t *shard, void *dcontext)
+{
+    shard->decode_cache = std::unique_ptr<decode_cache_t<opcode_data_t>>(
+        new decode_cache_t<opcode_data_t>(dcontext,
+                                          /*persist_decoded_instrs=*/false));
+}
+
+std::string
+opcode_mix_t::init_decode_cache(shard_data_t *shard, void *dcontext,
+                                offline_file_type_t filetype)
+{
+    make_decode_cache(shard, dcontext);
+    std::string err;
+    if (!TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, filetype)) {
```

Comment:
I see that opcde_mix sets the ISA mode for regdeps elsewhere: put a comment there that we want to move that to either this library or somewhere central?

Also move that code to here?